Build failure: python312Packages.pandas-stubs

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `python312.pkgs.pandas-stubs`

### Build log

```
=================================== FAILURES ===================================
______________________________ test_plot_scatter _______________________________

close_figures = None

    def test_plot_scatter(close_figures) -> None:
        check(
            assert_type(IRIS_DF.plot.scatter(x="SepalLength", y="SepalWidth"), Axes),
            Axes,
        )
        check(
            assert_type(
                IRIS_DF.plot(x="SepalLength", y="SepalWidth", kind="scatter"), Axes
            ),
            Axes,
        )
        check(
            assert_type(
>               IRIS_DF.plot.scatter(x="SepalLength", y="SepalWidth", subplots=True),
                npt.NDArray[np.object_],
            ),
            np.ndarray,
        )

tests/test_plotting.py:491: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/nix/store/jqw59f8hd0n99qq6hs8pcql054m4nnw5-python3.12-pandas-2.2.2/lib/python3.12/site-packages/pandas/plotting/_core.py:1748: in scatter
    return self(kind="scatter", x=x, y=y, s=s, c=c, **kwargs)
/nix/store/jqw59f8hd0n99qq6hs8pcql054m4nnw5-python3.12-pandas-2.2.2/lib/python3.12/site-packages/pandas/plotting/_core.py:975: in __call__
    return plot_backend.plot(data, x=x, y=y, kind=kind, **kwargs)
/nix/store/jqw59f8hd0n99qq6hs8pcql054m4nnw5-python3.12-pandas-2.2.2/lib/python3.12/site-packages/pandas/plotting/_matplotlib/__init__.py:71: in plot
    plot_obj.generate()
/nix/store/jqw59f8hd0n99qq6hs8pcql054m4nnw5-python3.12-pandas-2.2.2/lib/python3.12/site-packages/pandas/plotting/_matplotlib/core.py:503: in generate
    self._make_legend()
/nix/store/jqw59f8hd0n99qq6hs8pcql054m4nnw5-python3.12-pandas-2.2.2/lib/python3.12/site-packages/pandas/plotting/_matplotlib/core.py:896: in _make_legend
    ax.legend(loc="best")
/nix/store/8w0d3p71dg4myj6srw5al38773b02gr2-python3.12-matplotlib-3.9.0/lib/python3.12/site-packages/matplotlib/axes/_axes.py:341: in legend
    handles, labels, kwargs = mlegend._parse_legend_args([self], *args, **kwargs)
/nix/store/8w0d3p71dg4myj6srw5al38773b02gr2-python3.12-matplotlib-3.9.0/lib/python3.12/site-packages/matplotlib/legend.py:1364: in _parse_legend_args
    _api.warn_external(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

message = 'No artists with labels found to put in legend.  Note that artists whose label start with an underscore are ignored when legend() is called with no argument.'
category = None

    def warn_external(message, category=None):
        """
        `warnings.warn` wrapper that sets *stacklevel* to "outside Matplotlib".
    
        The original emitter of the warning can be obtained by patching this
        function back to `warnings.warn`, i.e. ``_api.warn_external =
        warnings.warn`` (or ``functools.partial(warnings.warn, stacklevel=2)``,
        etc.).
        """
        frame = sys._getframe()
        for stacklevel in itertools.count(1):
            if frame is None:
                # when called in embedded context may hit frame is None
                break
            if not re.match(r"\A(matplotlib|mpl_toolkits)(\Z|\.(?!tests\.))",
                            # Work around sphinx-gallery not setting __name__.
                            frame.f_globals.get("__name__", "")):
                break
            frame = frame.f_back
        # preemptively break reference cycle between locals and the frame
        del frame
>       warnings.warn(message, category, stacklevel)
E       UserWarning: No artists with labels found to put in legend.  Note that artists whose label start with an underscore are ignored when legend() is called with no argument.

/nix/store/8w0d3p71dg4myj6srw5al38773b02gr2-python3.12-matplotlib-3.9.0/lib/python3.12/site-packages/matplotlib/_api/__init__.py:381: UserWarning
=========================== short test summary info ============================
FAILED tests/test_plotting.py::test_plot_scatter - UserWarning: No artists with labels found to put in legend.  Note that arti...
=========== 1 failed, 982 passed, 7 skipped, 20 deselected in 19.88s ===========
```

### Notify maintainers
@malob 
recently edited: @fabaff 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.9.10, NixOS, 24.11 (Vicuna), 24.11pre-git`
 - multi-user?: `yes`
  - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.90.0`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/6pxw90lyfq5b52hq83d635lhdsgwjqk9-z9nih1pm3bycx3nib37ivdk0g0ds4akj-nixpkgs-src`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pandas-stubs/default.nix
+++ b/pkgs/development/python-modules/pandas-stubs/default.nix
@@ -1,9 +1,20 @@
 {
   lib,
   stdenv,
-  beautifulsoup4,
   buildPythonPackage,
+  pythonOlder,
   fetchFromGitHub,
+
+  # build-system
+  poetry-core,
+
+  # dependencies
+  numpy,
+  types-pytz,
+
+  # tests
+  pytestCheckHook,
+  beautifulsoup4,
   html5lib,
   jinja2,
   lxml,
@@ -11,16 +22,12 @@
   odfpy,
   openpyxl,
   pandas,
-  poetry-core,
   pyarrow,
   pyreadstat,
-  pytestCheckHook,
-  pythonOlder,
   scipy,
   sqlalchemy,
   tables,
   tabulate,
-  types-pytz,
   typing-extensions,
   xarray,
   xlsxwriter,
@@ -28,7 +35,7 @@
 
 buildPythonPackage rec {
   pname = "pandas-stubs";
-  version = "2.2.0.240218";
+  version = "2.2.2.240603";
   pyproject = true;
 
   disabled = pythonOlder "3.9";
@@ -37,17 +44,18 @@ buildPythonPackage rec {
     owner = "pandas-dev";
     repo = "pandas-stubs";
     rev = "refs/tags/v${version}";
-    hash = "sha256-416vyaHcSfTfkSNKZ05edozfsMmNKcpOZAoPenCLFzQ=";
+    hash = "sha256-wJ7lN1+BGNqhwgPLmmpb3foX6rIEy3R62rOvu0mr8xs=";
   };
 
-  nativeBuildInputs = [ poetry-core ];
+  build-system = [ poetry-core ];
 
-  propagatedBuildInputs = [
-    pandas
+  dependencies = [
+    numpy
     types-pytz
   ];
 
   nativeCheckInputs = [
+    pytestCheckHook
     beautifulsoup4
     html5lib
     jinja2
@@ -55,9 +63,9 @@ buildPythonPackage rec {
     matplotlib
     odfpy
     openpyxl
+    pandas
     pyarrow
     pyreadstat
-    pytestCheckHook
     scipy
     sqlalchemy
     tables
@@ -65,19 +73,28 @@ buildPythonPackage rec {
     typing-extensions
     xarray
     xlsxwriter
+
+    # missing dependencies in nixpkgs
+    # python-calamine
   ];
 
   disabledTests =
     [
-      # AttributeErrors, missing dependencies, error and warning checks
-      "test_types_groupby"
-      "test_frame_groupby_resample"
-      "test_orc"
-      "test_all_read_without_lxml_dtype_backend"
+      # Missing dependencies, error and warning checks
+      "test_all_read_without_lxml_dtype_backend" # pyarrow.orc
+      "test_orc" # pyarrow.orc
+      "test_read_excel" # python-calamine
+      "test_plotting" # UserWarning: No artists with labels found to put in legend.
+      "test_spss" # FutureWarning: ChainedAssignmentError: behaviour will change in pandas 3.0!
       "test_show_version"
     ]
     ++ lib.optionals stdenv.isDarwin [
-      "test_plotting" # Fatal Python error: Illegal instruction
+      "test_clipboard" # FileNotFoundError: [Errno 2] No such file or directory: 'pbcopy'
+    ]
+    ++ lib.optionals (stdenv.isDarwin && stdenv.isAarch64) [
+      # Disable tests for types that are not supported on aarch64 in `numpy` < 2.0
+      "test_astype_float" # `f16` and `float128`
+      "test_astype_complex" # `c32` and `complex256`
     ];
 
   pythonImportsCheck = [ "pandas" ];

